src/incremental.rs

```diff
@@ -1084,15 +1084,19 @@ impl State {
         unlock_or_activate!(automations, unlocked);
     }
 
-    pub fn buy_material(&mut self, id: ProductMaterialId, cnt: u32) {
-        for _ in 0..cnt {
-            let m = &mut self.materials[id];
-            if m.count() >= m.limit().unwrap_or(Count::MAX) {
-                continue;
-            }
-
-            let price = m.price();
-            if self.money >= price {
+    pub fn buy_material(&mut self, id: ProductMaterialId, cnt: u64) {
+    
+        let cnt_f64 = cnt as f64;
+        let m = &mut self.materials[id];
+        if m.count() >= m.limit().unwrap_or(Count::MAX) {
+            return; // Exit early if the material is already at the limit
+        }
+    
+        let price = m.price();
+        let total_cost = price * cnt_f64;
+    
+        if self.money >= total_cost {
+            for _ in 0..cnt {
                 m.buy();
                 self.money -= price;
```

Comment:
you can just do
```
self.money -= total_cost;
```
outside the for loop

Reply:
You´re right, should´ve thought about that.

I think I got confused there.

Reply:
Ah, don't worry, my code is confusing at this moment and not documented enough. 

Materials increase their price with each item bought(see `ProductMaterial::price`), so if you do the subtraction inside the for loop you might end up with negative amount of money, so we need the total_cost to be subtracted only once.

That's why I want to keep the old functionality also, as it buys up to `cnt` materials taking into account the everchanging price.

I will be thankful if you add comments for the two functions, but honestly don't worry about it as I need to sweep the code and add more comments wherever necessary(actually if you do have any confusions about the code you can leave some `TODO: add comment here` so I don't miss it).